test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1372 @@
+			] );
+	} );
+
+	// Previously we would unexpectedly duplicate the block on Enter.
+	test( 'should not multi select single block', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '1' );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await page.keyboard.press( 'Enter' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: '' } },
+			{ name: 'core/paragraph', attributes: { content: '' } },
+		] );
+	} );
+
+	test( 'should gradually multi-select', async ( {
+		page,
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/columns',
+			innerBlocks: [
+				{
+					name: 'core/column',
+					innerBlocks: [
+						{
+							name: 'core/paragraph',
+							attributes: { content: '1' },
+						},
+						{
+							name: 'core/paragraph',
+							attributes: { content: '2' },
+						},
+					],
+				},
+				{ name: 'core/column' },
+			],
+		} );
+		// Focus the last paragraph block.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Paragraph block' } )
+			.nth( 1 )
+			.click();
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toMatchObject( [
+				{ name: 'core/paragraph' },
+				{ name: 'core/paragraph' },
+			] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toMatchObject( [ { name: 'core/column' } ] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toMatchObject( [
+				{ name: 'core/column' },
+				{ name: 'core/column' },
+			] );
+
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect both columns to be deleted.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [ { name: 'core/columns', innerBlocks: [] } ] );
+	} );
+
+	test( 'should multi-select from within the list block', async ( {
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/list',
+			innerBlocks: [
+				{
+					name: 'core/list-item',
+					attributes: { content: '1' },
+				},
+			],
+		} );
+		// Focus on the list item.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Block: List item' } )
+			.getByRole( 'textbox' )
+			.click();
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
```

Comment:
Why do we need as many as 3 `select all` hits here?

Reply:
The first one selects the content of the list item. The second one selects the list block. The third one selects all blocks. I'll add a comment above to clarify!